server/client.go

```diff
@@ -2258,7 +2267,7 @@ func (c *client) processInboundClientMsg(msg []byte) {
 			atomic.LoadInt64(&c.srv.gateway.totalQSubs) > 0 {
 			collect = true
 		}
-		qnames = c.processMsgResults(c.acc, r, msg, c.pa.subject, c.pa.reply, collect)
+		qnames = c.processMsgResults(c.acc, r, msg, c.pa.subject, c.pa.reply, collect, false)
```

Comment:
This is accumulating arguments and bools are hard to reason about later on. This required?

Reply:
I still need several conditions. This function is quite big and complex and I feel like duplicating it would not be too wise. But we could have one with all params but not call directly, instead call functions with different names that call that one with appropriate params..

Reply:
I have created a `processMsgResultsEx()` and explicit functions that call that one with predefined params. I don't think you like the "Ex" functions, but then not sure what to do here.